Buildings/Resources/Scripts/travis/templates/core.py

```diff
@@ -0,0 +1,487 @@
+        type=str,
+        help="tool to run simulations (case-insensitive)",
+        default='dymola',
+        required=False,
+    )
+    parser.add_argument(
+        "--coverage",
+        type=float,
+        help="fraction of test coverage between 0 (>) and 1 (<=)",
+        default=1,
+        required=False,
+    )
+    parser.add_argument("--generate", help="generate combinations", action="store_true")
+    parser.add_argument("--simulate", help="path of combination file", action="store_true")
+    args = parser.parse_args()
+
+    assert (
+        args.coverage > 0 and args.coverage <= 1
+    ), "Fraction of test coverage must be between (>) 0 and (<=) 1."
+
+    return args
+
+
+def check_conf(model_name, simulator, conf=CONF):
+    """Check whether a model is included in conf.yml for the given simulator."""
+    simulator = simulator.lower()
+    for el in conf:
+        list_keys = list(el.keys())
+        if model_name == el['model_name'] and simulator in list(el.keys()):
+            return True
+    return False
+
+
+def simulate_case(arg, simulator):
+    """Set common parameters and run simulation with buildingspy.
+
+    Args:
+        arg: tuple[str, list[str], str]: Model name, list of class modifications, suffix for mat file name.
+        simulator: str: Modelica tool for simulating the model.
+
+    Returns:
+        tuple[int, str]: Error code, log.
+    """
+    # Local imports required for multiprocessing in Jupyter notebook.
+    import glob
+    import json
+    import os
+    import re
+    import shutil
+    import tempfile
+
+    simulator = simulator.lower()
+
+    if simulator == 'dymola':
+        from buildingspy.simulate.Dymola import Simulator
+    elif simulator == 'optimica':
+        from buildingspy.simulate.Optimica import Simulator
+    else:
+        return 4, f'Unsupported simulation tool: {simulator}.'
+
+    mat_root = re.split(r"\.", arg[0])[-1]
+    mat_suffix = re.sub(r"\.", "_", str(arg[2]))
+    output_dir_prefix = f"{mat_root}_{mat_suffix}"
+    cwd = os.getcwd()
+    # We need to create temporary directories at the same level as Buildings because of
+    # the way volumes are mounted in docker run, see Buildings/Resources/Scripts/travis/dymola/dymola.
+    output_dir_path = tempfile.mkdtemp(prefix=output_dir_prefix, dir=os.path.abspath(os.pardir))
+
+    # The following make Dymola worker cd into outputDirectory.
+    s = Simulator(arg[0], outputDirectory=output_dir_path)
+
+    if simulator == 'dymola':
+        s.addPreProcessingStatement(r'Advanced.TranslationInCommandLog:=true;')
+        s.addPreProcessingStatement(r'openModel("../Buildings/package.mo", changeDirectory=false);')
+    if simulator == 'optimica':
+        # Set MODELICAPATH (only in child process, so this won't affect main process).
+        os.environ['MODELICAPATH'] = os.path.abspath(os.pardir)
+
+    for modif in arg[1]:
+        s.addModelModifier(modif)
+
+    s.setSolver("CVode")
+    s.setTolerance(1e-6)
```

Comment:
Is it good to override the settings here rather than use the one from the `experiment` annotation, possibly overridden from `conf.yml`?